### ISSUE TITLE
Add an option for const enums generation

### ISSUE THREAD

**Issue by lxgreen** [stephenh/ts-proto#230]
As Typescript docs say, `const enum` members are inlined at use sites. For example, the following TS code

```
export enum DividerData_DividerType {
  DOUBLE = 'DOUBLE',
  SINGLE = 'SINGLE',
  DASHED = 'DASHED',
  DOTTED = 'DOTTED',
  UNRECOGNIZED = 'UNRECOGNIZED',
}

const divider = DividerData_DividerType.DASHED;
```

is translated to the following JS:

```
export var DividerData_DividerType;
(function (DividerData_DividerType) {
    DividerData_DividerType["DOUBLE"] = "DOUBLE";
    DividerData_DividerType["SINGLE"] = "SINGLE";
    DividerData_DividerType["DASHED"] = "DASHED";
    DividerData_DividerType["DOTTED"] = "DOTTED";
    DividerData_DividerType["UNRECOGNIZED"] = "UNRECOGNIZED";
})(DividerData_DividerType || (DividerData_DividerType = {}));

const divider = DividerData_DividerType.DASHED;
```

while the same example using `const enum`:

```
export const enum DividerData_DividerType {
  DOUBLE = 'DOUBLE',
  SINGLE = 'SINGLE',
  DASHED = 'DASHED',
  DOTTED = 'DOTTED',
  UNRECOGNIZED = 'UNRECOGNIZED',
}

const divider = DividerData_DividerType.DASHED;
```

is translated to just
```
const divider = "DASHED";
```

As you can see, this can have a dramatic effect on bundle size. While there are some cases when the enums have to be preserved in JS, they are not required in many cases.

**Comment by stephenh** [comment]
Sure, makes sense. I'd be tempted to just make enums always `const` but then at some point I'll get an issue filed that someone wants them _not_ `const`. :-)

Can you submit a PR with the new flag and an example of a mini proto file + `parameters.txt` in the `integration/` directory?

**Comment by stephenh** [comment]
Released as v1.71.0. Thanks!
